apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -871,77 +794,55 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
-    else
-      :empty_dataset ->
-        {:ok, nil}
-
-      %{acc: _, msg: error_message} ->
-        Logger.warning("Database error: #{error_message}.")
-        {:error, :database_error}
-
-      {:error, reason} ->
-        Logger.warning("Failed with reason: #{inspect(reason)}.")
-        {:error, :database_error}
+    with n when is_number(n) or is_nil(n) <- Repo.fetch_one(query, consistency: :quorum) do
+      {:ok, n}
     end
   end
 
-  def fetch_path_expiry(db_client, device_id, interface_descriptor, %Mapping{} = mapping, path)
+  def fetch_path_expiry(realm, device_id, interface_descriptor, %Mapping{} = mapping, path)
       when is_binary(device_id) and is_binary(path) do
-    # TODO: do not hardcode individual_properties here
-    fetch_property_value_statement = """
-    SELECT TTL(datetime_value)
-    FROM individual_properties
-    WHERE device_id=:device_id AND interface_id=:interface_id
-      AND endpoint_id=:endpoint_id AND path=:path
-    """
-
-    fetch_property_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(fetch_property_value_statement)
-      |> DatabaseQuery.put(:device_id, device_id)
-      |> DatabaseQuery.put(:interface_id, interface_descriptor.interface_id)
-      |> DatabaseQuery.put(:endpoint_id, mapping.endpoint_id)
-      |> DatabaseQuery.put(:path, path)
-      |> DatabaseQuery.consistency(:quorum)
-
-    with {:ok, result} <- DatabaseQuery.call(db_client, fetch_property_query),
-         ["ttl(datetime_value)": ttl] when is_integer(ttl) <- DatabaseResult.head(result) do
-      expiry_datetime =
-        DateTime.utc_now()
-        |> DateTime.to_unix()
-        |> :erlang.+(ttl)
-        |> DateTime.from_unix!()
+    %InterfaceDescriptor{interface_id: interface_id} = interface_descriptor
+    %Mapping{endpoint_id: endpoint_id} = mapping
+
+    keyspace_name =
+      CQLUtils.realm_name_to_keyspace_name(realm, Config.astarte_instance_id!())
+
+    q =
+      IndividualProperty
+      |> where(
+        device_id: ^device_id,
+        interface_id: ^interface_id,
+        endpoint_id: ^endpoint_id,
+        path: ^path
+      )
+      |> put_query_prefix(keyspace_name)
+      |> select([p], fragment("TTL(?)", p.reception_timestamp))
 
-      {:ok, expiry_datetime}
-    else
-      :empty_dataset ->
+    case Repo.all(q, consistency: :quorum) do
+      [] ->
         {:error, :property_not_set}
 
-      ["ttl(datetime_value)": nil] ->
+      [nil] ->
         {:ok, :no_expiry}
 
-      %{acc: _, msg: error_message} ->
-        Logger.warning("Database error: #{error_message}.")
-        {:error, :database_error}
+      [ttl] when is_integer(ttl) ->
+        expiry_datetime =
+          DateTime.utc_now()
+          |> DateTime.to_unix()
+          |> :erlang.+(ttl)
+          |> DateTime.from_unix!()
 
-      {:error, reason} ->
-        Logger.warning("Database error while retrieving property: #{inspect(reason)}.")
-        {:error, :database_error}
+        {:ok, expiry_datetime}
```

Comment:
Here errors are not handled anymore, Ecto crashes where we used to return `{:error, :database_error}`

Reply:
The call would have crashed also in the case of `{:error, :database_error}` because the result is not matched, see [here](https://github.com/astarte-platform/astarte/blob/7aebe4cd5dfde27b0ef5098fb7945f9320ac51ac/apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex#L603-L612) and [here](https://github.com/astarte-platform/astarte/blob/7aebe4cd5dfde27b0ef5098fb7945f9320ac51ac/apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex#L1015-L1027)